lms/djangoapps/instructor_task/rest_api/v1/views.py

```diff
@@ -78,11 +94,103 @@ def destroy(self, request, *args, **kwargs):
             return Response(status=status.HTTP_404_NOT_FOUND)
 
-        # update the task's status to REVOKED and then delete the task schedule instance
         task = schedule.task
+        # verify the task hasn't already been processed before revoking
+        try:
+            self._verify_task_state(task)
+        except TaskUpdateException as task_update_err:
+            return Response(
+                {"detail": str(task_update_err)},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        # update the task's status to REVOKED and then delete the task schedule instance
         log.info(f"Revoking instructor task with id '{task.id}' for course '{task.course_id}'")
         task.task_state = REVOKED
         task.save()
         schedule.delete()
 
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+    def patch(self, request, *args, **kwargs):
+        course_id = kwargs["course_id"]
+        schedule_id = kwargs["schedule_id"]
+
+        # extract schedule and email data from the request
+        schedule = request.data.get("schedule", None)
+        email_data = request.data.get("email", None)
+
+        try:
+            task_schedule = self._get_task_schedule_with_id(schedule_id)
+            task = task_schedule[0].task
+            self._verify_task_state(task)
+
+            with transaction.atomic():
+                if schedule:
+                    browser_timezone = request.data.get("browser_timezone", None)
+                    schedule_utc = convert_schedule_to_utc_from_local(schedule, browser_timezone, request.user)
+                    self._verify_valid_schedule(schedule_id, schedule_utc)
+                    task_schedule.update(task_due=schedule_utc)
+                if email_data:
+                    email_id = email_data.get("id")
+                    self._verify_task_and_email_associated(task, email_id)
+                    targets = email_data.get("targets")
+                    subject = email_data.get("subject")
+                    message = email_data.get("message")
+                    update_course_email(course_id, email_id, targets, subject, message)
+        except (TaskUpdateException, ValueError) as task_update_err:
+            return Response(
+                {"detail": str(task_update_err)},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+        except InstructorTaskSchedule.DoesNotExist as task_update_err:
+            return Response(
+                {"detail": str(task_update_err)},
+                status=status.HTTP_404_NOT_FOUND
+            )
+        else:
+            return Response(status=status.HTTP_200_OK)
+
+    def _get_task_schedule_with_id(self, schedule_id):
+        """
+        Verifies that there is an instructor task schedule instance that exists with the specified id.
+        """
+        task_schedule = InstructorTaskSchedule.objects.filter(pk=schedule_id)
+        if not task_schedule.exists():
```

Comment:
I was working with a queryset over getting the object instance as I was under the impression this is a more performant way to update the object instance. We will use the query set after its returned (if it exists) with an `update` call. 

For this reason I have this utility function to try and use the ORM to filter the queryset to the record we want and then check if there are any results (by invoking the `.exists()` call).